riscv/src/register/pmpcfgx.rs

```diff
@@ -13,6 +15,28 @@ pub enum Permission {
     RWX = 0b111,
 }
 
+impl TryFrom<u8> for Permission {
+    type Error = Error;
+
+    fn try_from(val: u8) -> Result<Self> {
+        match val {
+            0b000 => Ok(Self::NONE),
+            0b001 => Ok(Self::R),
+            0b010 => Ok(Self::W),
+            0b011 => Ok(Self::RW),
+            0b100 => Ok(Self::X),
+            0b101 => Ok(Self::RX),
+            0b110 => Ok(Self::WX),
+            0b111 => Ok(Self::RWX),
```

Comment:
what if valid values go from 0 to 6? I suggest using two different errors: `InvalidValue` (more generic) and `OutOfBounds` (taking into account the mask stuff). We can even define a different enum for typical `InvalidValue` types (`OutOfBounds`, `Unknown` or something like that)

Reply:
Not sure what you want here.

Currently, the `Error::InvalidValue` variant is used to signify an invalid bitfield value for a register.

`Error::OutOfBounds` is used to indicate an invalid index access outside an acceptable range.

Are you suggesting the the `Error::InvalidValue` variant itself be a wrapper around a `enum InvalidValue` type? I think this complicates things too much, if that's what you're suggesting.